go/vt/sqlparser/token.go

```diff
@@ -334,8 +334,16 @@ func (tkn *Tokenizer) Scan() (int, []byte) {
 		switch ch {
 		case eofChar:
 			return 0, nil
-		case '=', ',', ';', '(', ')', '+', '*', '%', '&', '|', '^', '~':
+		case '=', ',', ';', '(', ')', '+', '*', '%', '&', '^', '~':
 			return int(ch), nil
+		case '|':
+			switch tkn.lastChar {
```

Comment:
not sure which is more go-ish, but in simple cases like this i feel like i've more often seen it as:
```go
if tkn.lastChar == '|' {
  tkn.next()
  return OR, nil
}
return int(ch), nil
```